dydx/dydxViews/dydxViews/Shared/dydxNumberFieldView.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  dydxNumberFieldView.swift
+//  dydxViews
+//
+//  Created by Michael Maguire on 7/18/24.
+//
+
+import SwiftUI
+import dydxFormatter
+import PlatformUI
+
+/// Effectively a TextField which forces its input as a number
+struct dydxNumberField: View {
```

Comment:
would like to use this as a more SwiftUI-friendly version of PlatformInput going fwd 

@ruixhuang lmk if you have objections